iwyu_ast_util.h

```diff
@@ -96,11 +108,19 @@ class ASTNode {
       : kind_(kTypelocKind), as_typeloc_(typeloc),
         parent_(nullptr), in_fwd_decl_context_(false) { }
   ASTNode(const clang::NestedNameSpecifier* nns)
-      : kind_(kNNSKind), as_nns_(nns),
-        parent_(nullptr), in_fwd_decl_context_(false) { }
+      : kind_(kNNSKind),
+        as_nns_(nns),
+        parent_(nullptr),
+        in_fwd_decl_context_(false) {
+    CHECK_(*nns);
+  }
   ASTNode(const clang::NestedNameSpecifierLoc* nnsloc)
-      : kind_(kNNSLocKind), as_nnsloc_(nnsloc),
-        parent_(nullptr), in_fwd_decl_context_(false) { }
+      : kind_(kNNSLocKind),
+        as_nnsloc_(nnsloc),
+        parent_(nullptr),
+        in_fwd_decl_context_(false) {
+    CHECK_(nnsloc->getNestedNameSpecifier());
```

Comment:
Hoping that it is always `true`. It would be good to hear from @mizvekov whether null NNS may appear in the AST or not.

Reply:
Since the comment is outdated I can't look at the specific code this was talking about.

But a NNS of kind null appears in the same places a nullptr NNS * would before my patch, when the type in question was written with no / empty name qualifier.

Reply:
Ah, sorry! IWYU has [explicit checks](https://github.com/include-what-you-use/include-what-you-use/pull/1793/files#diff-7ca107c8b836c1765e8b735b74f06833bec48f308a76017886cee1d5736002d9R476) that NNS is not null before constructing `ASTNode`, hence a null NNS should simply not get here.